### PR TITLE
add constraint to service create ref

### PR DESCRIPTION
**Pull request by sfsmithcha**
Add description of constraints to service_create.md based upon @dongluochen description for https://github.com/docker/docker/pull/24397

Signed-off-by: Charles Smith charles.smith@docker.com

### CODE REVIEW

**Comment by sfsmithcha** [comment]
@dongluochen , @thaJeztah PTAL

File: docs/reference/commandline/service_create.md [hunk 1]
```diff
@@ -145,6 +145,32 @@ The following command creates a "global" service:
 $ docker service create --name redis_2 --mode global redis:3.0.6
 ```
 
+### Service constraint
```
Comment (thaJeztah) [inline docs/reference/commandline/service_create.md]: constraints? (plural)

**Comment by thaJeztah** [comment]
one nit/suggestion, but LGTM otherwise, thanks!

File: docs/reference/commandline/service_create.md [hunk 2]
```diff
@@ -145,6 +145,32 @@ The following command creates a "global" service:
...
+
+| node attribute | matches | example |
+|:------------- |:-------------| :---------------------------------------------|
+| node.id | node ID | `node.id == 2ivku8v2gvtg4`                               |
+| node.hostname | node hostname | `node.hostname != node-2`                    |
+| node.role | node role: manager | `node.role == manager`                      |
+| node.labels | user defined node labels | `node.labels.security == high`      |
+| engine.labels | Docker Engine's labels | `engine.labels.operatingsystem == ubuntu 14.04`|
+
+`engine.labels` apply to Docker Engine attributes like operating system,
```
Comment (dongluochen) [inline docs/reference/commandline/service_create.md]: `Docker Engine attributes` -> `Docker Engine labels`. `labels` are widely used term in Docker daemon.

File: docs/reference/commandline/service_create.md [hunk 3]
```diff
@@ -145,6 +145,32 @@ The following command creates a "global" service:
...
+drivers, etc. Swarm administrators add `node.labels` for operational purposes.
```
Comment (dongluochen) [inline docs/reference/commandline/service_create.md]: "operational purposes" -> "operational purposes by using `docker node update` command"

**Comment by dongluochen** [comment]
thanks @sfsmithcha. The change on constraints has not vendored into docker/docker yet. We should merge this after vendor done.

**Comment by sfsmithcha** [comment]
Thanks, @dongluochen . Let me know when.

**Comment by aaronlehmann** [comment]
@dongluochen: Actually it looks like it has been vendored.

**Comment by thaJeztah** [comment]
I'll merge this; we know for sure it'll be there before 1.12GA

**Pull request merged**